### PR TITLE
45813 - Consultation Status Date Trigger AB#45813

### PR DESCRIPTION
**Pull request by csmith-he**
Consultation Status function updated so that a status is created/changed as triggered by the creation of a new Consultation resource or the creation/update of the completion date

[AB#45813](https://hedev.visualstudio.com/76e71f30-b1ad-438a-a096-89ac98712f4e/_workitems/edit/45813) in Keystone backlog

### CODE REVIEW

File: arches_her/functions/consultation_status_function.py [hunk 1]
```diff
@@ -1,130 +1,144 @@
...
+            date_comp_tile = None
+            if tile.nodegroup_id == str(date_comp_node.nodegroup_id):
+                    date_comp_tile = tile
+            else:
+                date_comp_tile_filter = Tile.objects.filter(resourceinstance_id=tile.resourceinstance_id,nodegroup_id=date_comp_node.nodegroup_id)
+                if len(date_comp_tile_filter) > 0:
+                    date_comp_tile = date_comp_tile_filter[0]
+            if  date_comp_tile != None:
+                date_comp_node_string = str(date_comp_node.nodeid)
+                date_comp_value = date_comp_tile.data[date_comp_node_string]
```
Comment (aj-he) [inline arches_her/functions/consultation_status_function.py:79]: Should you be using a datatype factory to get the value?

File: arches_her/functions/consultation_status_function.py [hunk 2]
```diff
@@ -1,130 +1,144 @@
...
 
-class ConsultationStatusFunction(BaseFunction): 
+class ConsultationStatusFunction(BaseFunction):
+
+    def handle_boolean_tile(self,tile,nodegroupid,nodeid,status,create):
+        try:
+            nodeid_str = str(nodeid)
+            cons_status_tile = None
+
+            if create == True:
```
Comment (aj-he) [inline arches_her/functions/consultation_status_function.py:32]: if consultation status is a 1:1 then do we ever need to specify to create it? Should is just either fetch it or create it?

File: arches_her/functions/consultation_status_function.py [hunk 3]
```diff
@@ -1,130 +1,144 @@
...
+            if tile.nodegroup_id == str(date_comp_node.nodegroup_id):
+                    date_comp_tile = tile
+            else:
+                date_comp_tile_filter = Tile.objects.filter(resourceinstance_id=tile.resourceinstance_id,nodegroup_id=date_comp_node.nodegroup_id)
+                if len(date_comp_tile_filter) > 0:
+                    date_comp_tile = date_comp_tile_filter[0]
+            if  date_comp_tile != None:
+                date_comp_node_string = str(date_comp_node.nodeid)
+                date_comp_value = date_comp_tile.data[date_comp_node_string]
+                date_comp = datetime.datetime.strptime(date_comp_value, "%Y-%m-%d").date()
```
Comment (aj-he) [inline arches_her/functions/consultation_status_function.py:80]: what happens it the logged and target are set but the completed on node is null/empty? This line might error and you probably won't want to continue as you need this value.

**Review by aj-he — CHANGES_REQUESTED** [review]
A few things to to confirm. Also, there seems to be a load of report template changes that are included. Can you check this please.

File: arches_her/functions/consultation_status_function.py (hunk 2 above)
Comment (csmith-he) [reply arches_her/functions/consultation_status_function.py:32]: I did try using Tile.objects.get_or_create initially but it didn't work and I didn't quite get to the bottom of why which is why I used this method

File: arches_her/functions/consultation_status_function.py (hunk 1 above)
Comment (csmith-he) [reply arches_her/functions/consultation_status_function.py:79]: Ok, I'll have a look at using that method

File: arches_her/functions/consultation_status_function.py (hunk 3 above)
Comment (csmith-he) [reply arches_her/functions/consultation_status_function.py:80]: I'll alter the code to handle this sort of issue

**Pull request closed without merging**